tests/test_integration_layout.py

```diff
@@ -0,0 +1,188 @@
+            (SCREEN_W - 2 - 1, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 1 because of #root's border
+            1,
+        ],
+        [
+            SCREEN_SIZE,
+            1,
+            "border: solid white;",  # #root has a visible border
+            "align: center top;",  # placeholders are centered horizontally
+            # #root's virtual size=screen size
+            (SCREEN_W, SCREEN_H),
+            # placeholders width=same than screen, minus 2 borders :: height=default height
+            (SCREEN_W - 2, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 1 because of #root's border
+            1,
+        ],
+        [
+            SCREEN_SIZE,
+            4,
+            "border: solid white;",  # #root has a visible border
+            "align: center top;",  # placeholders are centered horizontally
+            # #root's virtual height should be as high as its stacked content
+            (SCREEN_W - 2 - 1, PLACEHOLDERS_DEFAULT_H * 4),
+            # placeholders width=same than screen, minus 2 borders, minus scrollbar :: height=default height
+            (SCREEN_W - 2 - 1, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 1 because of #root's border
+            1,
+        ],
+    ),
+)
+async def test_composition_of_vertical_container_with_children(
+    screen_size: Size,
+    placeholders_count: int,
+    root_container_style: str,
+    placeholders_style: str,
+    expected_placeholders_size: tuple[int, int],
+    expected_root_widget_virtual_size: tuple[int, int],
+    expected_placeholders_offset_x: int,
+    event_loop: asyncio.AbstractEventLoop,
+):
+    class VerticalContainer(Widget):
+        CSS = (
+            """
+        VerticalContainer {
+            layout: vertical;
+            overflow: hidden auto;
+            ${root_container_style}
+        }
+        
+        VerticalContainer Placeholder {
+            height: ${placeholders_height};
+            ${placeholders_style}
+        }
+        """.replace(
+                "${root_container_style}", root_container_style
+            )
+            .replace("${placeholders_height}", str(PLACEHOLDERS_DEFAULT_H))
+            .replace("${placeholders_style}", placeholders_style)
+        )
+
+    class MyTestApp(AppTest):
+        def compose(self) -> ComposeResult:
+            placeholders = [
+                Placeholder(id=f"placeholder_{i}", name=f"Placeholder #{i}")
+                for i in range(placeholders_count)
+            ]
+
+            yield VerticalContainer(*placeholders, id="root")
+
+    app = MyTestApp(size=screen_size, test_name="compositor")
+
+    expected_screen_size = Size(*screen_size)
+
+    # On Windows the screen size is always 1 cell smaller than the requested dimensions.
+    # Let's take this into account into our "expected_*" measurements:
+    if WINDOWS:
```

Comment:
This is only the case on legacy windows, which we won't be supporting. So we can set `legacy_windows=False` on Windows, and we won't need this conditional.